File: scrudbeans-integration-tests-java/src/main/java/myjavapackage/model/Order.java
package myjavapackage.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.manosbatsis.scrudbeans.api.mdd.annotation.model.ScrudBean;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Entity
@Table(name = "product_orders")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
@ScrudBean(dtoTypeNames = "myjavapackage.dto.OrderUpdateCommentDTO")
@Schema(name = "Order", description = "A model representing an order of product items")
public class Order {

    @Id
    @GeneratedValue(generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid2")
    private String id;

    @NotNull
    @Column(nullable = false)
    @Schema(description = "The client's email", required = true)
    private String email;

    @Column(length = 512)
    @Schema(description = "Oder comment", required = false)
    private String comment;

    @CreatedDate
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    @Schema(description = "Date created", readOnly = true)
    @Column(name = "date_created", nullable = false, updatable = false)
    private LocalDateTime createdDate;

    @LastModifiedDate
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    @Schema(description = "Date last modified", readOnly = true)
    @Column(name = "date_last_modified", nullable = false)
    private LocalDateTime lastModifiedDate;

}

File: scrudbeans-integration-tests-kotlin/src/main/kotlin/mykotlinpackage/model/Order.kt
package mykotlinpackage.model

import com.github.manosbatsis.scrudbeans.api.annotation.model.ScrudBean
import com.github.manosbatsis.scrudbeans.model.AbstractAuditableEntity
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EntityListeners
import jakarta.persistence.Table
import jakarta.validation.constraints.NotNull
import mykotlinpackage.service.CustomBaseService
import mykotlinpackage.service.CustomBaseServiceImpl
import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder
import org.apache.commons.lang3.builder.ToStringBuilder
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.util.*

@Entity
@Table(name = "product_orders")
@EntityListeners(AuditingEntityListener::class)
@ScrudBean(
    serviceSuperInterface = CustomBaseService::class,
    serviceImplSuperClass = CustomBaseServiceImpl::class,
)
@Schema(name = "Order", description = "A model representing an order of product items")
class Order(
    id: UUID? = null,
    @field:NotNull
    @field:Column(nullable = false)
    @field:Schema(title = "The client's email", required = true)
    var email: String,

    @field:Column(length = 512)
    @field:Schema(title = "Order comment", required = false)
    var comment: String? = null,
) : AbstractAuditableEntity(id) {
    override fun equals(obj: Any?): Boolean {
        if (obj == null) return false
        if (obj === this) return true
        if (!Order::class.isInstance(obj)) return false
        obj as Order
        return EqualsBuilder()
            .appendSuper(super.equals(obj))
            .append(email, obj.email)
            .append(comment, obj.comment)
            .isEquals
    }

    override fun hashCode(): Int {
        return HashCodeBuilder(19, 39)
            .appendSuper(super.hashCode())
            .append(email)
            .append(comment)
            .toHashCode()
    }

    override fun toString(): String {
        return ToStringBuilder(this)
            .appendSuper(super.toString())
            .append("email", email)
            .append("comment", comment)
            .build()
    }
}
